adafruit_ble/services/standard/device_info.py

```diff
@@ -65,11 +64,19 @@ def __init__(
             if model_number is None:
                 model_number = sys.platform
             if serial_number is None:
-                serial_number = binascii.hexlify(
-                    microcontroller.cpu.uid  # pylint: disable=no-member
-                ).decode("utf-8")
+                try:
+                    import microcontroller  # pylint: disable=import-outside-toplevel
+
+                    serial_number = binascii.hexlify(
+                        microcontroller.cpu.uid  # pylint: disable=no-member
+                    ).decode("utf-8")
+                except ImportError:
+                    pass
             if firmware_revision is None:
-                firmware_revision = os.uname().version
+                try:
+                    firmware_revision = os.uname().version
+                except AttributeError:
+                    pass
```

Comment:
```suggestion
                firmware_revision = getattr(os.uname(), "version", None)
```

Reply:
It's `os.uname()` that doesn't exist on Windows, not `version`.

Reply:
Is this correct then? Should it be `if hasattr(os, "uname")`?

Reply:
It could be a `hasattr()` test, but the exception works fine and is less code, because I'd have to test for `uname`, and then do the call. "Ask for forgiveness instead of permission", as the Python saying goes
```python
Python 3.8.2 (default, Apr 27 2020, 15:53:34) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.foo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'os' has no attribute 'foo'
``